.github/workflows/main.yml

```diff
@@ -279,7 +279,7 @@ jobs:
       run: |
         cd build/
         tar cvjf ${{ steps.extract_branch.outputs.branch }}.tar.bz2 doc_doxygen/
-        python -m awscli s3 cp ./${{ steps.extract_branch.outputs.branch }}.tar.bz2 s3://xgboost-docs/ --acl public-read
+        python -m awscli s3 cp ./${{ steps.extract_branch.outputs.branch }}.tar.bz2 s3://xgboost-docs/doxygen/ --acl public-read
```

Comment:
ReadTheDocs doesn't have Doxygen installed on their build server, so here we build Doxygen in GitHub Actions and then upload the tarball to the `xgboost-docs` S3 bucket. The sphinx build script expects to find the tarball at a fixed location. See https://github.com/dmlc/xgboost/blob/b9a4f3336af7c78f9f7b66f7c91a419108b7ec83/doc/conf.py#L36-L55

Reply:
Unrelated to this PR.  But the `call` to shell code can be completely replaced by Python.